### ISSUE TITLE
pip install failed on linux cluster

### ISSUE THREAD

**Issue by wanting0wang** [google/sentencepiece#403]
System Info:
Linux version 4.14.0-115.7.1.el7a.ppc64le (mockbuild@ppc-056.build.eng.bos.redhat.com) (gcc version 4.8.5 20150623 (Red Hat 4.8.5-36) (GCC))

I tried both installing from PyPI and installing from source file, but neither of them worked.

When installing from PyPI:
```
$ pip install sentencepiece
Collecting sentencepiece
  Using cached https://files.pythonhosted.org/packages/1b/87/c3c2fa8cbec61fffe031ca9f0da512747520bec9be7f886f748457daac31/sentencepiece-0.1.83.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-t33o0yz4/sentencepiece/setup.py", line 29, in <module>
        with codecs.open(os.path.join('..', 'VERSION'), 'r', 'utf-8') as f:
      File "/opt/anaconda3/lib/python3.6/codecs.py", line 897, in open
        file = builtins.open(filename, mode, buffering)
    FileNotFoundError: [Errno 2] No such file or directory: '../VERSION'

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-install-t33o0yz4/sentencepiece/
```

I then manually downloaded the tar.gz source file, uncompressed it, changed the directory to "./python", and tried to install directly from the setup.py:
```
$ python setup.py install
Package sentencepiece was not found in the pkg-config search path.
Perhaps you should add the directory containing `sentencepiece.pc'
to the PKG_CONFIG_PATH environment variable
No package 'sentencepiece' found
Failed to find sentencepiece pkgconfig
```

However `pip install .` gives a different error message:
```
$ pip install .
Processing <...>/sentencepiece-0.1.83/python
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-req-build-209jgy5x/setup.py", line 29, in <module>
        with codecs.open(os.path.join('..', 'VERSION'), 'r', 'utf-8') as f:
      File "/opt/anaconda3/lib/python3.6/codecs.py", line 897, in open
        file = builtins.open(filename, mode, buffering)
    FileNotFoundError: [Errno 2] No such file or directory: '../VERSION'

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-req-build-209jgy5x/
```

Does anyone know what might be wrong and how to fix it?
Thank you!

**Comment by bpben** [comment]
Using [this process](https://www.kaggle.com/sunhwan/google-sentencepiece) worked for me on Red Hat Linux Server 7.7.

**Comment by npow** [comment]
Just ran into this as well. `pip install sentencepiece==0.1.82` works though

**Comment by leezu** [comment]
Worked for me after upgrading to pip 20.0.2

**Comment by DaveXanatos** [comment]
Same error here.  Raspberry Pi 4B (4 gig), Python 3.7, Raspbian Buster

```
$ pip install sentencepiece
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting sentencepiece
  Using cached https://files.pythonhosted.org/packages/1b/87/c3c2fa8cbec61fffe031ca9f0da512747520bec9be7f886f748457daac31/sentencepiece-0.1.83.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-6tdniw95/sentencepiece/setup.py", line 29, in <module>
        with codecs.open(os.path.join('..', 'VERSION'), 'r', 'utf-8') as f:
      File "/usr/lib/python3.7/codecs.py", line 898, in open
        file = builtins.open(filename, mode, buffering)
    FileNotFoundError: [Errno 2] No such file or directory: '../VERSION'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-install-6tdniw95/sentencepiece/

```
Also tried earlier versions ( pip install sentencepiece==0.1.83 ) to same effect.

Need this to run other stuff in my project....  help appreciated very much.

**Comment by cagansevencan** [comment]
I have this same error, tried to install sentenpiece manually still no changes...

**Comment by DaveXanatos** [comment]
Haven't heard anything back myself, have put it on hold while I work on visual systems/networking.  I hope that with all the stuff going on in the world and everybody being stuck at home, SentencePiece might get some love & attention 😊

 

Stay healthy

**Comment by Kefas** [comment]
Same error here:
Linux raspberrypi 4.19.66-v7+ #1253 SMP Thu Aug 15 11:49:46 BST 2019 armv7l GNU/Linux
cmake version 3.13.4

tried both pip and installing from the source. Nothing works
I've been trying to install v0.1.83

**Comment by DaveXanatos** [comment]
We're all waiting on the SentencePiece Dev Crew apparently.  Sure could use a code fix so I can get GPT-2 running on my Raspberries…

 

I hope everybody's safe & healthy.

**Comment by ochafik** [comment]
I had the same issue on MacOS X (python 3.8.2 + pip 20.0.2).

I downgraded to python 3.7.7 (using brew-installed pyenv to switch easily), reupdated pip (`pip install --upgrade pip`) and it worked without a hiccup (it picked up version 0.1.85).

Good luck & be safe!

**Comment by DaveXanatos** [comment]
I'm already on Python 3.7 and it still won't go, but I'm using Debian, not a Mac…  Glad you got it to go.  Hopefully the rest of us can get some relief soon 😊

 

Stay healthy! 😊

**Comment by taku910** [comment]
Could you try the latest release?

Latest python package gets self-contained so it should be compiled as long as C++ lib is installed in the system.

**Comment by kaoutar55** [comment]
I am running into the same issue on a power pc environment. I have Python 3.7.7 on redhat linux.

 pip install sentencepiece
Collecting sentencepiece
  Using cached sentencepiece-0.1.91.tar.gz (500 kB)
    ERROR: Command errored out with exit status 1:
     command: /gpfs/u/home/ADCS/ADCSkmkh/scratch-shared/miniconda3/envs/transformers/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-yr40y_3v/sentencepiece/setup.py'"'"'; __file__='"'"'/tmp/pip-install-yr40y_3v/sentencepiece/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-install-yr40y_3v/sentencepiece/pip-egg-info
         cwd: /tmp/pip-install-yr40y_3v/sentencepiece/
    Complete output (5 lines):
    Package sentencepiece was not found in the pkg-config search path.
    Perhaps you should add the directory containing `sentencepiece.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'sentencepiece' found
    Failed to find sentencepiece pkgconfig
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.

**Comment by bobbyheykal** [comment]
I got exacly the same error while building from source using Raspberry pi 4,  either on raspbian or ubuntu. Is there any update?

**Comment by Nolence** [comment]
This still feels like an issue. Consider reopening? Running into this on a Jetson Nano with python 3.6

```
$ pip install sentencepiece

Collecting sentencepiece
  Using cached sentencepiece-0.1.91.tar.gz (500 kB)
    ERROR: Command errored out with exit status 1:
     command: /home/user/Documents/example/venv/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-4drgkry3/sentencepiece/setup.py'"'"'; __file__='"'"'/tmp/pip-install-4drgkry3/sentencepiece/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-_q8z_uh9
         cwd: /tmp/pip-install-4drgkry3/sentencepiece/
    Complete output (5 lines):
    Package sentencepiece was not found in the pkg-config search path.
    Perhaps you should add the directory containing `sentencepiece.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'sentencepiece' found
    Failed to find sentencepiece pkgconfig
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

**Comment by DaveXanatos** [comment]
Still an issue for me.  It seems there is one and only one, specific configuration of dependencies between about 50 total things that this works on…  I've given up on this one and moved on unfortunately.

 

Dave

**Comment by jshamg** [comment]
Got the same issue, tried dozens of versions, also tried building from source. Something is broken in the fallback package(the source). All whl packages work as far as I know but on arm and other platforms except x86 the fallback is used. Please fix this ASAP. Transformers do not work on arm because of that. 
Cheers!

**Comment by jshamg** [comment]
@taku910 why did you close this? scentencepiece is unusable on sbc's with arm arch. Your package is the reason why transformers wont install on all of these devices. Fix it!

**Comment by DaveXanatos** [comment]
> @taku910 .... Your package is the reason why transformers wont install on all of these devices. Fix it!

PLEASE!  I have raw GPT-2 running just fine, but anything requiring sentencepiece blows up.  There's a LOT of us out here who would really like to see this fixed.  Thanks.

**Comment by jshamg** [comment]
I'm gonna open a new Bug and reference this one since this one hasn't been addressed.
@DaveXanatos maybe you can comment on that one, too so we get more attention by the team to fix it.

Cheers!

**Comment by taku910** [comment]
Releases v0.1.93 that supports ppc64le, arm, and s390x, it will soon available via pypi.

https://github.com/google/sentencepiece/releases/tag/v0.1.93

**Comment by pmadruga** [comment]
@taku910 which is the correct wheel for aarch64?

**Comment by simaattar** [comment]
Un Ubuntu, first run :
`sudo apt-get install cmake build-essential pkg-config libgoogle-perftools-dev`
and then:
`sudo pip install sentencepiece`

**Comment by qiopta** [comment]
Thanks @simaattar I've been looking for a couple of days and this Unbuntu fix is simple and effective!

**Comment by DaveXanatos** [comment]
Thanks @simaattar - I'm on a Debian repo but I recognize all of those and believe that they may work for me as well.  Will advise.
